core/common/byte_stream.cpp

```diff
@@ -0,0 +1,14 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common/byte_stream.hpp"
+
+OUTCOME_CPP_DEFINE_CATEGORY(kagome::common, ByteStream::AdvanceErrc, e) {
+  using kagome::common::ByteStream;
+  switch (e) {
+    case ByteStream::AdvanceErrc::OUT_OF_BOUNDARIES:
+      return "Advance cannot move pointer outside of boundaries";
+  }
```

Comment:
In our documentation there is a note 
"    default: return "unknown"; // NOTE: do not forget to handle everything else "
And I agree with this approach, as it may be helpful in cases when a new constant was added but forgotten to be processed in switch, or the value is just invalid.